src/hotspot/share/compiler/compilerDefinitions.cpp

```diff
@@ -117,19 +117,14 @@ intx CompilerConfig::scaled_freq_log(intx freq_log, double scale) {
   // of length InvocationCounter::number_of_count_bits. Mask values are always
   // one bit shorter then the value of the notification frequency. Set
   // max_freq_bits accordingly.
-  intx max_freq_bits = InvocationCounter::number_of_count_bits + 1;
+  int max_freq_bits = InvocationCounter::number_of_count_bits + 1;
```

Comment:
Why was the type changed?

Reply:
Presumably the type change is so MIN2 can be used on line 127.  And there's no obvious reason why the bit counts here should be intx.

Reply:
The reason is that ilog2 returns int. I incorrectly reasoned that ilog2 would use the same return type as the input argument.